common/shell/shell_platform.c

```diff
@@ -0,0 +1,607 @@
+/*
+    NAME: PLATFORM COMMAND
+    FILE: shell_platform.c
+    DESCRIPTION: OEM commands including gpio, sensor relative function access.
+    AUTHOR: MouchenHung
+    DATE/VERSION: 2022.01.11 - v1.0.0
+    CHIP/OS: AST1030 - Zephyr
+    Note:
+    (1) User command table 
+          [topic]               [description]               [support]   [command]
+        * GPIO                                              O
+            * List group        List gpios' info in group   o           platform gpio list_group <gpio_device>
+            * List all          List all gpios' info        o           platform gpio list_all
+            * List multifnctl   List multi-fn-ctl regs      o           platform gpio multifnctl
+            * Get               Get one gpio info           o           platform gpio get <gpio_num>
+            * Set value         Set one gpio value          o           platform gpio set val <gpio_num> <value>
+            * Set direction     Set one gpio direction      x           TODO
+            * Set INT type      Set one gpio interrupt T    o           platform gpio set int_type <gpio_num> <type>
+        * SENSOR                                            O
+            * List all          List all sensors' info      o           platform sensor list_all
+            * Get               Get one sensor info         o           platform sensor get <sensor_num>
+            * Set polling en    Set one sensor polling      x           TODO
+            * Set mbr           Set one sensor mbr          x           TODO
+            * Set threshold     Set one sensor threshold    x           TODO
+        * I2C SLAVE                                         X
+
+    (2) Some hard code features need to be modified if CHIP is different
+
+    (3) GPIO List all/Get/Set value/Set direction are not protected by PINMASK_RESERVE_CHECK. It might cause system-hang problem!
+
+    (4) "List multifnctl" command only lists relative registers mentioned in chip spec
+        chapter "5.Multi-function Pins Mapping and Control" 
+*/
+
+#include <zephyr.h>
+#include <sys/printk.h>
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include <logging/log.h>
+#include <shell/shell.h>
+#include <device.h>
+#include <devicetree.h>
+
+/* Include GPIO */
+#include <drivers/gpio.h>
+#include "plat_gpio.h"
+#include "hal_gpio.h"
+
+/* Include SENSOR */
+#include "sensor.h"
+
+/* Include config settings */
+#include "shell_platform.h"
+
+/* Declare Log */
+LOG_MODULE_REGISTER(shell_platform);
+
+/* Declare Code info */
+#define RELEASE_VERSION "v1.0.0"
```

Comment:
Should this version info move to the location where other release version info for BIC is kept?
Second question is do we want to have a separate shell version or use the main BIC version?

Reply:
About the RELEASE_VERSION above is just my own coding style that giving every small project a version, I will delete it if not necessary.